src/SFML/Graphics/Image.cpp

```diff
@@ -260,44 +260,31 @@ std::optional<Image> Image::loadFromStream(InputStream& stream)
 
 
 ////////////////////////////////////////////////////////////
-bool Image::saveToFile(const std::filesystem::path& filename) const
+bool Image::saveToFile(const std::filesystem::path& filename, SaveFormat format) const
 {
     // Make sure the image is not empty
     if (!m_pixels.empty() && m_size.x > 0 && m_size.y > 0)
     {
-        // Deduce the image type from its extension
+        const Vector2i convertedSize = Vector2i(m_size);
 
-        // Extract the extension
-        const std::filesystem::path extension     = filename.extension();
-        const Vector2i              convertedSize = Vector2i(m_size);
-
-        if (extension == ".bmp")
-        {
-            // BMP format
-            if (stbi_write_bmp(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data()))
-                return true;
-        }
-        else if (extension == ".tga")
-        {
-            // TGA format
-            if (stbi_write_tga(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data()))
-                return true;
-        }
-        else if (extension == ".png")
-        {
-            // PNG format
-            if (stbi_write_png(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data(), 0))
-                return true;
-        }
-        else if (extension == ".jpg" || extension == ".jpeg")
-        {
-            // JPG format
-            if (stbi_write_jpg(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data(), 90))
-                return true;
-        }
-        else
+        switch (format)
         {
-            err() << "Image file extension " << extension << " not supported\n";
+            case SaveFormat::BMP:
+                if (stbi_write_bmp(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data()))
+                    return true;
+                break;
+            case SaveFormat::TGA:
+                if (stbi_write_tga(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data()))
+                    return true;
+                break;
+            case SaveFormat::PNG:
+                if (stbi_write_png(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data(), 0))
+                    return true;
+                break;
+            case SaveFormat::JPG:
+                if (stbi_write_jpg(filename.string().c_str(), convertedSize.x, convertedSize.y, 4, m_pixels.data(), 90))
+                    return true;
+                break;
```

Comment:
Despite being exhaustive, we'd still need to handle the default case, as `static_cast<sf::Image::SaveFormat>(42)` will compile just fine.

Reply:
If anything we ought to `assert` against that. It's a pretty unambiguous bug if you someone manage to use an invalid enum value. If you want we can add a `default` case with that assertion.

Reply:
I think an assert should be fine, then we wouldn't need a default case.

Reply:
I'm not sure where to add that `assert` without adding a `default` case which I was hoping to avoid. Where are you picturing we add the `assert`?

Reply:
I'd have put them at the beginning of the function like any other argument checks. If you pass in invalid argument values, the function should fail in precondition asserts.